platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/render/RenderTestActivity.java

```diff
@@ -0,0 +1,314 @@
+    mapSnapshotter.start(result -> {
+      Bitmap snapshot = result.getBitmap();
+      imageView.setImageBitmap(snapshot);
+      renderResultMap.put(renderTestDefinition, snapshot);
+      if (renderResultMap.size() != testSize) {
+        continueTesting(renderTestDefinition);
+      } else {
+        finishTesting();
+      }
+    }, error -> Timber.e(error));
+  }
+
+  private void continueTesting(RenderTestDefinition renderTestDefinition) {
+    int next = renderTestDefinitions.indexOf(renderTestDefinition) + 1;
+    Timber.d("Next test: %s / %s", next, renderTestDefinitions.size());
+    render(renderTestDefinitions.get(next), renderTestDefinitions.size());
+  }
+
+  private void finishTesting() {
+    new SaveResultToDiskTask(onSnapshotReadyListener, renderResultMap).execute();
+  }
+
+  //
+  // Save tests results to disk
+  //
+  private static class SaveResultToDiskTask extends AsyncTask<Void, Void, Void> {
+
+    private OnSnapshotReadyListener onSnapshotReadyListener;
+    private Map<RenderTestDefinition, Bitmap> renderResultMap;
+
+    SaveResultToDiskTask(OnSnapshotReadyListener onSnapshotReadyListener,
+                         Map<RenderTestDefinition, Bitmap> renderResultMap) {
+      this.onSnapshotReadyListener = onSnapshotReadyListener;
+      this.renderResultMap = renderResultMap;
+    }
+
+    @Override
+    protected Void doInBackground(Void... voids) {
+      if (isExternalStorageWritable()) {
+        try {
+          File testResultDir = FileUtils.createTestResultRootFolder();
+          String basePath = testResultDir.getAbsolutePath();
+          for (Map.Entry<RenderTestDefinition, Bitmap> testResult : renderResultMap.entrySet()) {
+            writeResultToDisk(basePath, testResult);
+          }
+        } catch (final Exception exception) {
+          Timber.e(exception);
+        }
+      }
+      return null;
+    }
+
+    private void writeResultToDisk(String path, Map.Entry<RenderTestDefinition, Bitmap> result) throws IOException {
+      RenderTestDefinition definition = result.getKey();
+      String categoryName = definition.getCategory();
+      String categoryPath = String.format("%s/%s", path, categoryName);
+      FileUtils.createCategoryDirectory(categoryPath);
+      String testName = result.getKey().getName();
+      String testDir = FileUtils.createTestDirectory(categoryPath, testName);
+      FileUtils.writeTestResultToDisk(testDir, result.getValue());
+    }
+
+    private boolean isExternalStorageWritable() {
+      return Environment.MEDIA_MOUNTED.equals(Environment.getExternalStorageState());
+    }
+
+    @Override
+    protected void onPostExecute(Void aVoid) {
+      super.onPostExecute(aVoid);
+      if (onSnapshotReadyListener != null) {
+        onSnapshotReadyListener.onSnapshotReady();
+      }
+    }
+  }
+
+  //
+  // Callback configuration to notify test executor of test finishing
+  //
+  public interface OnSnapshotReadyListener {
```

Comment:
This interface's name and method's name are a bit ambiguous. Could we rename it to something along the lines of `OnTestSnapshotsReadyListener`?